SimG4CMS/Calo/src/CaloSD.cc

```diff
@@ -111,43 +116,38 @@ CaloSD::CaloSD(G4String name, const DDCompactView & cpv,
 }
 
 CaloSD::~CaloSD() { 
-  if (slave)           delete slave; 
-  if (theHC)           delete theHC;
-  if (meanResponse)    delete meanResponse;
+  delete slave; 
+  delete theHC;
+  delete meanResponse;
 }
 
-bool CaloSD::ProcessHits(G4Step * aStep, G4TouchableHistory * ) {
-  
-  NaNTrap( aStep ) ;
+bool CaloSD::ProcessHits(G4Step * aStep, G4TouchableHistory * tHistory) {
   
-  if (aStep == NULL) {
-    return true;
-  } else {
-    if (getStepInfo(aStep)) {
-      if (hitExists() == false && edepositEM+edepositHAD>0.) 
-        currentHit = createNewHit();
-    }
+  NaNTrap(aStep);
+  theTouchableHistory = tHistory;
+  if(getStepInfo(aStep) && !hitExists() && edepositEM+edepositHAD>0.f) {
```

Comment:
This was never needed and likely is left in many SD classes due to historical reasons. 

There is also NaNTrap - I am not sure how it is useful, I personally prefer if FPE arithmetic checks are enabled on compiler level in jenkins tests and RelVal tests, home made NaNTrap looks strange, however, in this PR it is simplified, so cost much less than before.